en/debug/eclipse_jlink.md

```diff
@@ -78,6 +78,20 @@ For more information, see: [https://gnu-mcu-eclipse.github.io/debug/jlink/instal
    ![Eclipse: GDB Segger Debug config: debugger tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_debugger_tab.png)
    ![Eclipse: GDB Segger Debug config: startup tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_startup_tab.png)
 
+## SEGGER Task-aware debugging
+Also known as Thread Aware debugging allows you to show the context of all running threads/tasks instead of just the stack current task. This is quite usefull since PX4 tend run many different tasks.
+
+1. Make sure that in your build the following config is enabled *Enable TCBinfo struct for debug* symbol `CONFIG_DEBUG_TCBINFO`
+   - Hint you can change this through make <target> menuconfig
+   ![NuttX: Menuconfig: CONFIG_DEBUG_TCBINFO](../../assets/debug/nuttx_tcb_task_aware.png)
+1. Run `make jlink-nuttx` to compile the **jlink-nuttx.so** library
+1. Modify Eclipse: GDB Segger Debug config *Other options* to include `-rtos /home/<PX4 path>/Tools/jlink-nuttx.so`
+   ![Eclipse: GDB Segger Debug config RTOS aware: debugger tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_task_aware.png)
+1. When running the debugger you should see now multiple threads instead of 1
+   ![Eclipse: GDB Segger Debug config RTOS aware: debug session](../../assets/debug/eclipse_settings_debug_config_gdb_segger_task_aware_tasks.png)
+
+
+
```

Comment:
Minor tidy:

```suggestion
To enable this feature for use in Eclipse:

1. You first need to enable `CONFIG_DEBUG_TCBINFO` in the NuttX configuration for your build (to expose the TCB offsets).
    - Open a terminal in the root of your PX4-Autopilot source code
    - In the terminal, open `menuconfig` using the appropriate make target for the build.
      This will be something like:
      ```
      make px4_fmu-v5_default boardguiconfig
      ```
      (See [PX4 Menuconfig Setup](../hardware/porting_guide_config.md#px4-menuconfig-setup) for more information) on using the config tools).
   - Ensure that the *Enable TCBinfo struct for debug* is selected as shown:
      ![NuttX: Menuconfig: CONFIG_DEBUG_TCBINFO](../../assets/debug/nuttx_tcb_task_aware.png)
1. Compile the **jlink-nuttx.so** library in the terminal by running the following command in the terminal:
   ```
   make jlink-nuttx
   ```
1. Modify Eclipse to use this libary.
    In the *J-Link GDB Server Setup* configuration, update **Other options** to include `-rtos /home/<PX4 path>/Tools/jlink-nuttx.so`, as shown in the image below.

   ![Eclipse: GDB Segger Debug config RTOS aware: debugger tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_task_aware.png)
1. When running the debugger you should see now multiple threads instead of just one:

   ![Eclipse: GDB Segger Debug config RTOS aware: debug session](../../assets/debug/eclipse_settings_debug_config_gdb_segger_task_aware_tasks.png)

```